### PR TITLE
Issue 376: Update buildbot_graph_builder so it can take any list of jobs

### PR DESCRIPTION
**Pull request by adusca**
buildbot_graph_builder will take a list of jobs that we want to trigger and
a revision, and create a graph with the necessary build jobs.

r? @armenzg

### CODE REVIEW

File: mozci/sources/buildbot_bridge.py [hunk 1]
```diff
@@ -118,13 +118,42 @@ def buildbot_graph_builder(builders):
...
     """
     graph = {}
+
+    # We need to determine what upstream jobs need to be triggered besides the
+    # builders already on our list
     for b in builders:
-        graph[b] = None
+        # for test jobs, determine_trigger_objective[0] is the build
+        # job when no build job exists the test job when the build job
+        # is completed and None when the build job is running
```
Comment (armenzg) [inline mozci/sources/buildbot_bridge.py]: Could you please structure this paragraph as a bullet point?
It is a bit hard to read and understand.

File: mozci/sources/buildbot_bridge.py [hunk 2]
```diff
@@ -118,13 +118,42 @@ def buildbot_graph_builder(builders):
...
+        if is_downstream(b):
+            objective = determine_trigger_objective(revision, b)[0]
+
+            if objective == b:
```
Comment (armenzg) [inline mozci/sources/buildbot_bridge.py:140]: Please add a note in hear to understand what we're doing.
Same below.

**Comment by armenzg** [comment]
Looks good to me. Please merge when you address the nits mentioned.

**Comment by adusca** [comment]
Thanks for the review! I hope the new comments make more sense.

**Pull request merged**